src/Payfast.php

```diff
@@ -165,7 +184,10 @@ public function buildForm()
         $htmlForm = '<form id="payfast-pay-form" action="https://'.$this->host.'/eng/process" method="post">';
         foreach($this->vars as $name => $value)
         {
-            $htmlForm .= '<input type="hidden" name="'.$name.'" value="'.$value.'">';
+            // empty fields should not be sent across it breaks certain payment methods
+            if (!empty($value) || $value === 0) {
```

Comment:
For subscriptions, the PayFast API does not like it if empty fields are sent, as it messes up their signature validation. But value 0 should be passed for the `cycle` option on subscriptions